app/main.py

```diff
@@ -2,5 +2,16 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_storage: dict = {}
+
+    def internal(*args, **kwargs) -> Callable:
+        argument_list: list = [args, kwargs]
+        for cached_result, cached_arguments in cache_storage.items():
+            if argument_list == cached_arguments:
+                print("Getting from cache")
+                return cached_result
+        print("Calculating new result")
+        result = func(*args, **kwargs)
+        cache_storage[result] = argument_list
```

Comment:
The keys in the dictionary should correspond to the function's arguments, and the values should represent the function's output. This structure allows for efficient retrieval and mapping of function results, making it easy to access the outcomes based on specific inputs.

Reply:
Okay, but that wasn't stated in the task. Updated.